### PR TITLE
Simplify examples

### PR DESCRIPTION
**Pull request by surgura**
- Added and reworked many examples. See the README file in the examples directory.
- Most examples now don't work with a database. Database is optional to flatten learning curve.
- Added helper functions and classes for things occurring often, such as batch creation and fitness functions. Many things are just a lot more straightforward as customizable stuff is hidden behind defaults(see e.g. random number generation).

### CODE REVIEW

File: standard_resources/revolve2/standard_resources/simulation.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+STANDARD_SIMULATION_TIME = 30
+STANDARD_SAMPLING_FREQUENCY = 0.0001
+STANDARD_SIMULATION_TIMESTEP = 0.001
+STANDARD_CONTROL_FREQUENCY = 60
+
+
+def create_batch_single_robot_standard(
+    robot: ModularRobot,
+    terrain: Terrain,
+    simulation_time: Optional[int] = STANDARD_SIMULATION_TIME,
```
Comment (tckarenchiang) [inline standard_resources/revolve2/standard_resources/simulation.py:22]: is it intended not to expose `control_frequency` as in `create_batch_multiple_isolated_robots_standard`?

File: examples/evaluate_multiple_isolated_robots/main.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+from revolve2.standard_resources import fitness_functions, modular_robots
+from revolve2.standard_resources import terrains as standard_terrains
+from revolve2.standard_resources.rng import make_rng
+from revolve2.standard_resources.simulation import (
+    create_batch_multiple_isolated_robots_standard,
+)
+
+
+def main() -> None:
+    """Run the simulation."""
```
Comment (tckarenchiang) [inline examples/evaluate_multiple_isolated_robots/main.py:28]: missing standard logging setup here?

File: examples/evaluate_single_robot/main.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+from revolve2.core.modular_robot import ModularRobot, get_body_states_single_robot
+from revolve2.core.modular_robot.brains import BrainCpgNetworkNeighborRandom
+from revolve2.runners.mujoco import LocalRunner
+from revolve2.standard_resources import fitness_functions, modular_robots, terrains
+from revolve2.standard_resources.rng import make_rng
+from revolve2.standard_resources.simulation import create_batch_single_robot_standard
+
+
+def main() -> None:
+    """Run the simulation."""
```
Comment (tckarenchiang) [inline examples/evaluate_single_robot/main.py:21]: same here

File: examples/simple_ea_xor_database/main.py [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+Introduction:
+In the previous example, all results were written to the console.
+We would like to store all interesting data we come across in a more structured way.
+There are many options, such as saving text to files, various kinds of databases, or others such as the 'weights and biases' service.
+Here we choose to use the SQLite database.
+This a local relational database stored in a single file that is designed to be as simple as possible.
+It is very beneficial to have a basic understanding of relational databases, as this tutorial will not explain them in-depth.
+
+To simplify working with this database, we will use to 'object-relational mapping' (ORM) library 'SQLAlchemy'.
+This software lets you declare the database structure in Python, simply by creating classes simular to Python's 'dataclass'.
```
Comment (tckarenchiang) [inline examples/simple_ea_xor_database/main.py:22]:
```suggestion
This software lets you declare the database structure in Python, simply by creating classes similar to Python's 'dataclass'.
```

File: examples/simple_ea_xor_database/main.py [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+            session.add(generation)
+            session.commit()
+
+
+def main() -> None:
+    """Run the program."""
+    # Set up standard logging.
+    setup_logging(file_name="log.txt")
+
+    # Open the database, only if it does not already exists.
```
Comment (tckarenchiang) [inline examples/simple_ea_xor_database/main.py:234]:
```suggestion
    # Open the database, only if it does not already exist.
```

**Comment by tckarenchiang** [comment]
I just left some comments. I think the database separation and logging facility are very nice enhancements :+1:

File: standard_resources/revolve2/standard_resources/simulation.py (hunk 1 above)
Comment (surgura) [reply standard_resources/revolve2/standard_resources/simulation.py:22]: I considered it not very important, but to be complete I will add it.

File: examples/evaluate_multiple_isolated_robots/main.py (hunk 2 above)
Comment (surgura) [reply examples/evaluate_multiple_isolated_robots/main.py:29]: It's not really needed there but to not have students copy paste code and forget to initialize logging I added it to all examples.

File: examples/simulate_single_robot/main.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    # You probably don't have to worry about this, but if you are interested, take a look inside the function.
+    batch = create_batch_single_robot_standard(robot=robot, terrain=terrains.flat())
+
+    # Create a runner that will perform the simulation.
+    # This tutorial chooses to use Mujoco, but your version of revolve might contain other simulators as well.
+    runner = LocalRunner()
+    # Here we use the runner to run a batch.
+    # Once a runner finishes a batch, it can be reused to run a new batch.
+    # (We only run one batch in this tutorial, so we only use the runner once.)
+    asyncio.run(runner.run_batch(batch))
```
[the remark this answers is not in the document]
Comment (surgura) [reply examples/simulate_single_robot/main.py:81]: I also run Wayland and I don't get that warning and I don't think I have seen it on other machines. I guess if it works it does not really matter. If it turns out it does we will hear soon enough from the students :v: In any case this part of the code is a copy paste from the past year so it should be ok.

**Pull request merged**